cherwell_connector.py

```diff
@@ -134,418 +179,461 @@ def _make_rest_call(self, endpoint, action_result, headers=None, params=None, da
+            response = requests.post(endpoint, data=data, headers=headers, timeout=self._timeout, verify=self._verify)
+        except Exception as e:
+            error_message = self._get_error_message_from_exception(e)
+            self.debug_print("Error to make request call Error:{0}".format(error_message))
 
-    def _get_customer_recid(self, action_result, full_name):
-        ret_val, busobid_cid = self._get_busobid(action_result, 'CustomerInternal')
+        if response.status_code != 200:
+            if refresh_token:
+                # Request for fetching new token from refresh token failed, so try fetching new token forcefully
+                return self._get_oauth_token(action_result=action_result, force_new_token=True)
+            else:
+                return RetVal(action_result.set_status(phantom.APP_ERROR, "Error fetching token"), response.text)
+
+        response_json = response.json()
+
+        self._state["access_token"] = response_json["access_token"]
+        self._state["refresh_token"] = response_json["refresh_token"]
+
+        return RetVal(action_result.set_status(phantom.APP_SUCCESS, "Successfully fetched token"), self._state.get("access_token"))
+
+    def _encrypt_state(self, state):
+        if 'access_token' in state and 'refresh_token' in state:
+            access_token = state['access_token']
+            refresh_token = state['refresh_token']
+            try:
+                state['access_token'] = encryption_helper.encrypt(  # pylint: disable=E1101
+                    json.dumps(access_token),
+                    self.asset_id
+                )
+                state['refresh_token'] = encryption_helper.encrypt(  # pylint: disable=E1101
+                    json.dumps(refresh_token),
+                    self.asset_id
+                )
+                state[IS_STATE_ENCRYPTED] = True
+            except Exception as e:
+                error_message = self._get_error_message_from_exception(e)
+                self.debug_print("Unable to make encryption Error:{}".format(error_message))
+        return state
+
+    def _decrypt_state(self, state):
+        if state.get(IS_STATE_ENCRYPTED, False):
+            return state
+        if 'access_token' in state and 'refresh_token' in state:
+            try:
+                access_token = encryption_helper.decrypt(  # pylint: disable=E1101
+                    state['access_token'],
+                    self.asset_id
+                )
+                refresh_token = encryption_helper.decrypt(  # pylint: disable=E1101
+                    state['refresh_token'],
+                    self.asset_id
+                )
+                state['access_token'] = json.loads(access_token)
+                state['refresh_token'] = json.loads(refresh_token)
+            except Exception as e:
+                error_message = self._get_error_message_from_exception(e)
+                self.debug_print("Unable to make decryption. Error:{0}".format(error_message))
```

Comment:
Can we change the message to "Error occurred while decrypting the token: {}".format(error_message)?